examples/js/exporters/GLTFExporter.js

```diff
@@ -1028,6 +1004,73 @@ THREE.GLTFExporter.prototype = {
 
 			}
 
+			var forceIndices = options.forceIndices;
+			var isMultiMaterial = Array.isArray( mesh.material );
+
+			if ( ! forceIndices && geometry.index === null && isMultiMaterial ) {
+
+				// temporal workaround.
+				console.warn( 'THREE.GLTFExporter: Creating index for non-indexed multi-material mesh.' );
+				forceIndices = true;
+
+			}
+
+			var forcedIndex = false;
+
+			if ( geometry.index === null && forceIndices ) {
+
+				var indices = new Uint32Array( geometry.attributes.position.count );
+
+				for ( var i = 0, il = indices.length; i < il; i ++ ) {
+
+					indices[ i ] = i;
+
+				}
+
+				geometry.index = new THREE.Uint32BufferAttribute( indices, 1 );
```

Comment:
minor: is it preferred to use `geometry.setIndex( foo )`?

Reply:
Oh, I didn't know `.setIndex()` method exists. Updated, thanks.

I haven't replaced `index` reference with `.getIndex()` because it seems `.getIndex()` isn't used even in core.